src/components/AffiliateBanner.js

```diff
@@ -0,0 +1,113 @@
+import React, {useEffect} from 'react';
+import '../css/AffiliateBanner.css';
+import LogoImg from '../images/soulsmile-1024.png';
+import ComputerImg from '../images/computer_affiliate_version.png';
+import Button from 'react-bootstrap/Button';
+
+import GirlfriendImg from '../images/product_girlfriend_img.jpeg';
+import ByHumankindImg from '../images/product_byhumankind_img.jpeg';
+import CicadaImg from '../images/product_cicada_img.jpeg';
+import RocketbookImg from '../images/product_rocketbook_img.jpeg';
+
+var REACT_APP_AIRTABLE_RETAILERS_DOC = process.env.REACT_APP_AIRTABLE_RETAILERS_DOC;
+
+function AffiliateBanner(props) {
+
+    const { collab } = props;
+    //URLSearchParams Attempt
+    // //change above: function AffiliateBanner({ location }, props)
+    // function useQuery() {
+    //     return new URLSearchParams(location.search);
+    // }
+
+    // let query = useQuery();
+    // var affiliate = parseFloat(query.get("affiliate"));
+
+    // if(affiliate == 'girlfriend') {
+    //     collab = 'Girlfriend Collective';
+    // }
+
+    const [adRetailer, setAdRetailer] = React.useState([]);
+    
+    
+    useEffect(() => {
+        fetch(REACT_APP_AIRTABLE_RETAILERS_DOC)
+			.then(res => res.json())
+			.then(res => {
+                const data = res.records;
+                data.sort(function (a, b) {
+                    return ((a["fields"]["Name"] < b["fields"]["Name"]) ? -1 : (a["fields"]["Name"] > b["fields"]["Name"]) ? 1 : 0);
+                });
+
+                //Set product_img
+                var product_img;
+                switch(collab) {
+                    case 'Girlfriend Collective':
+                        product_img = GirlfriendImg;
+                        break;
+                    case 'by Humankind':
+                        product_img = ByHumankindImg;
+                        break;
+                    case 'Cicada':
+                        product_img = CicadaImg;
+                        break;
+                    case 'Rocketbook':
+                        product_img = RocketbookImg;
+                        break;
+                    default:
+                        product_img = null;
+                }
+                
+                var currentAdRetailer = [];
+                for (var j = 0; j < data.length; j++) {
```

Comment:
If you're only looking to set one retailer, it seems like iterating through all retailers in the Airtable would add a lot of extra time when starting up this page, especially if we have a ton of retailers eventually. Maybe we could instead insert the retailerID into the link (and pass it into the props) instead of the retailer name, or find a way to index on the retailer name in the results from the Airtable?

Reply:
This is a good point, but if/when we make the change where we store everything in the AirTable (rather than the "simple" implementation we currently use where some things are basically hardcoded), how could we access something like a company's image from the Airtable without looping like this? The Airtable didn't seem O(1) access to me in this sense but maybe I am misunderstanding something about how it works.